assignment2/q1.py

```diff
@@ -0,0 +1,36 @@
+# Print AncestorsGiven a Binary Tree and a key, 
+# write a function that prints all the ancestors of the key in the givenbinary tree.
+class Node:
+ 
+    
+    def __init__(self, data):
+        self.data = data 
+        self.left = None
+        self.right = None
+
+    def printAncestors(root, target):
+
+        if root == None:
+            return False
+
+        if root.data == target:
+            return True
+
+
+        if (printAncestors(root.left, target) or printAncestors(root.right, target)):
+            print root.data,
+            return True
+
+        return False
+
+    
```

Comment:
Maybe you can put the test code into a method called "test_ancestors" or something, and then call test_ancestors if the file is executed? Or perhaps just add unittests XD.